core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -101,14 +112,13 @@ protected String name() {
     Preconditions.checkArgument(namespace.levels().length >= 1,
         "Missing database in table identifier: %s", namespace);
 
-    Joiner slash = Joiner.on("/");
-    Path nsPath = new Path(slash.join(warehouseLocation, slash.join(namespace.levels())));
+    Path nsPath = new Path(SLASH.join(warehouseLocation, SLASH.join(namespace.levels())));
```

Comment:
This shouldn't use the outer `SLASH.join`. Instead, use the `new Path(String, String)` constructor.

Reply:
fixed